tools/vscode/stl.code-workspace

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+{
+    "folders": [
+        {
+            "path": "../../stl"
+        }
+    ],
+    "settings": {
+        "editor.formatOnSave": true,
+        "files.associations": {
+            "**/stl/inc/**": "cpp"

```

Comment:
don't we already do this someplace?

Also can't we do all of this in the local .vscode settings files without dealing with workspaces?

Reply:
> don't we already do this someplace?

You're thinking of the MSVC-internal `src/vctools/NonShip/vscode/stdhpp.code-workspace` and friends, which haven't appeared in GitHub - until now!

> Also can't we do all of this in the local .vscode settings files without dealing with workspaces?

I believe that would be a separate change. Currently, we gitignore all `.vscode` directories:

https://github.com/microsoft/STL/blob/d9cf06ead0f12479c2a6fce3135ae521d870df55/.gitignore#L5

It's possible to add gitignored files to source control, which is a little weird. My understanding is that this effectively provides a "default" file, which can then be customized by the user (without appearing as a pending change).

I don't know if VSCode looks for a `.vscode` directory in the opened folder, or if it also searches parent directories. We can add a `.vscode` folder to the root of the repo, but if the behavior is the former, it'll only help people opening the entire repo as a folder (which, as I described to Billy, results in an experience that I find undesirable - but others may like). If the behavior is the latter, I agree that it would supersede this `stl.code-workspace` as currently written.

The main advantage of `*.code-workspace` files is the ability to express multi-root workspaces, but I currently avoid that due to microsoft/vscode-cpptools#3285 which interferes with clang-format.

Reply:
I just tested the behavior, and VSCode looks for `.vscode` in the opened folder only - not recursively through parents. So we can add a `.vscode` in the root, but it won't be helpful to me or anyone else trying to open subdirectories.

Since I have no objections to having both, I will add this to the PR.

Reply:
Ooh, I can exclude vcpkg! I think that eliminates my desire to have a separate workspace.